### PR TITLE
Avoid every execution and allocation

### PR DESCRIPTION
**Pull request by aimwhy**
```javascript

export default function compose(...funcs) {
  if (!funcs.length) {
    return (...args) => args[0]
  } else {
    const last = funcs[funcs.length - 1]
    const rest = funcs.slice(0, -1)
    return (...args) => rest.reduceRight((composed, f) => f(composed), last(...args))
  }
}
```

### CODE REVIEW

File: src/compose.js [hunk 1]
```diff
@@ -10,14 +10,15 @@
  */
  
 export default function compose(...funcs) {
+  const isEmpty = funcs.length === 0
```
Comment (gaearon) [inline src/compose.js]: Let’s do it the other way: return another function early if there are no `funs`.

```js
export default function compose(...funcs) {
  if (!funcs.length) {
    return (...args) => args[0];
  }

  // ...

}
```

Then we can remove those ternaries and conditions below.

**Comment by aimwhy** [comment]
>Expected indentation of 4 space characters but found 2  ?
> Mixed spaces and tabs  ? 


Error: The Travis CI build could not complete due to an error  


Too young for me  and too simple for my think  ( ▼-▼ )

File: src/compose.js [hunk 2]
```diff
@@ -10,14 +10,12 @@
...
-    const last = funcs[funcs.length - 1]
-    const rest = funcs.slice(0, -1)
-
-    return rest.reduceRight((composed, f) => f(composed), last(...args))
+  if(funcs.length){
+    const last = funcs[funcs.length - 1];
+    const rest = funcs.slice(0, -1);
+    
+    return (...args) => rest.reduceRight((composed, f) => f(composed), last(...args));
+  }else{
```
Comment (gaearon) [inline src/compose.js]: Let’s flip the condition and make the `0` case an early return.

```js
if (!...) {
  return ...
}

...
```

Comment (aimwhy) [inline src/compose.js]: Just started using github
why The Travis CI build Report an error always ?
i dont use local editor ,only use explore
ok, put commented

**Comment by gaearon** [comment]
I don’t understand why linter crashes either :disappointed:

**Comment by gaearon** [comment]
Thanks!

**Comment by dd1994** [comment]
too young, too simple, 这也能膜 🌝

**Pull request merged**